### PR TITLE
Remove vertx-hazelcast dependency

### PR DESCRIPTION
**Pull request by tsegismont**
Fixes vert-x3/issues#416

### CODE REVIEW

File: vertx-service-discovery/src/test/java/io/vertx/servicediscovery/impl/DiscoveryImplClusteredTest.java [hunk 1]
```diff
@@ -19,21 +19,26 @@
...
+public class DiscoveryImplClusteredTest extends DiscoveryImplTestBase {
 
   @Before
   public void setUp() {
-    Vertx.clusteredVertx(new VertxOptions().setClusterHost("127.0.0.1"), ar -> {
+    FakeClusterManager.reset();
+    VertxOptions options = new VertxOptions()
+      .setClusterManager(new FakeClusterManager())
+      .setClusterHost("127.0.0.1");
+    Vertx.clusteredVertx(options, ar -> {
```
Comment (cescoffier) [inline vertx-service-discovery/src/test/java/io/vertx/servicediscovery/impl/DiscoveryImplClusteredTest.java]: I'm a bit worried we don't test the same thing anymore. Typically, with a "Real" cluster manager we have latency in the replication, here it's instantaneous. Of course, it should not matter, but in practice, it tends to matter.

The existing approach was far from perfect, and on a real cluster, the latency and replication error can be terrible.

Ideally, we would need a _wiremock_ for cluster managers where we can adapt the condition and response to be sure the code still behave correctly.

**Comment by tsegismont** [comment]
This test is executed in the HZ and ISPN cluster manager modules (not with
the fake CM of course)

**Comment by vietj** [comment]
@cescoffier perhaps we can introduce random latencies in the fake CM to make to improve ?

**Comment by cescoffier** [comment]
@vietj Yes, that would be great:

* inject latency
* failures

So we can be sure we handle these conditions the right way. 

That probably something that should be tracked in its own issue. @tsegismont what do you think?

**Comment by cescoffier** [comment]
BTW, merging.

**Comment by tsegismont** [comment]
@cescoffier sounds like a good idea

**Pull request merged**